From d918d16f03e657e50ec995bbc59fbdc9c4be2511 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Thu, 20 Mar 2025 11:13:29 +0000
Subject: [PATCH 07/10] move asm/unaligned.h to linux/unaligned.h

This commit accommodates commit 5f60d5f6bbc1("move asm/unaligned.h to 
linux/unaligned.h") which move asm/unaligned.h to linux/unaligned.h.
This commit was merged in the v6.12 development cycle and causes 
compilation failures with the v6.12 kernel, which are listed below.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=5f60d5f6bbc1

Resolved compilation errors:

src/ice_fw_update.c:4:10: fatal error: asm/unaligned.h: No such file or directory
    4 | #include <asm/unaligned.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/ice_fw_update.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ice_fw_update.c b/src/ice_fw_update.c
index e617487..2a2f6d7 100644
--- a/src/ice_fw_update.c
+++ b/src/ice_fw_update.c
@@ -1,7 +1,7 @@
 /* SPDX-License-Identifier: GPL-2.0-only */
 /* Copyright (C) 2018-2024 Intel Corporation */
 
-#include <asm/unaligned.h>
+#include <linux/unaligned.h>
 #include <linux/uuid.h>
 #include <linux/crc32.h>
 #if IS_ENABLED(CONFIG_PLDMFW)
-- 
2.47.1

